File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/download/DownloadTaskListenerAdapter.java
package com.ashlikun.okhttputils.http.download;

/**
 * 作者　　: 李坤
 * 创建时间: 15:10 Administrator
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：实现接口的抽象类
 */
public abstract class DownloadTaskListenerAdapter implements DownloadTaskListener {
    /**
     * 下载中
     *
     * @param completedSize
     * @param totalSize
     * @param percent
     * @param downloadTask
     */
    @Override
    public void onDownloading(DownloadTask downloadTask, long completedSize, long totalSize, double percent) {

    }

    /**
     * 下载暂停
     *
     * @param downloadTask
     * @param completedSize
     * @param totalSize
     * @param percent
     */
    @Override
    public void onPause(DownloadTask downloadTask, long completedSize, long totalSize, double percent) {

    }

    /**
     * 下载取消
     *
     * @param downloadTask
     */
    @Override
    public void onCancel(DownloadTask downloadTask) {

    }


    /**
     * 下载失败
     *
     * @param downloadTask
     * @param errorCode    {@link DownloadStatus}
     */
    @Override
    public void onError(DownloadTask downloadTask, int errorCode) {

    }


}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/download/DownloadTaskListenerAdapter.kt
package com.ashlikun.okhttputils.http.download

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.17 17:34
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：实现接口的抽象类
 */
abstract class DownloadTaskListenerAdapter : DownloadTaskListener {
    /**
     * 下载中
     *
     * @param completedSize
     * @param totalSize
     * @param percent
     * @param downloadTask
     */
    override fun onDownloading(
        downloadTask: DownloadTask,
        completedSize: Long,
        totalSize: Long,
        percent: Double
    ) {
    }

    /**
     * 下载暂停
     *
     * @param downloadTask
     * @param completedSize
     * @param totalSize
     * @param percent
     */
    override fun onPause(
        downloadTask: DownloadTask,
        completedSize: Long,
        totalSize: Long,
        percent: Double
    ) {
    }

    /**
     * 下载取消
     *
     * @param downloadTask
     */
    override fun onCancel(downloadTask: DownloadTask) {}

    /**
     * 下载失败
     *
     * @param downloadTask
     * @param errorCode    [DownloadStatus]
     */
    override fun onError(downloadTask: DownloadTask, errorCode: Int) {}
}